benchmarks/launch_latency/launch_latency.cpp

```diff
@@ -0,0 +1,259 @@
+#include <Kokkos_Core.hpp>
+
+template <int N>
+struct TestFunctor {
+  double values[N];
+  Kokkos::View<double*> a;
+  int K;
+  TestFunctor(Kokkos::View<double*> a_, int K_) : a(a_), K(K_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i) const {
+    for (int j = 0; j < K; j++) a(i) += 1.0 * i * values[j];
+  }
+};
+
+template <int N>
+struct TestRFunctor {
+  double values[N];
+  Kokkos::View<double*> a;
+  int K;
+  TestRFunctor(Kokkos::View<double*> a_, int K_) : a(a_), K(K_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, double& lsum) const {
+    for (int j = 0; j < K; j++) a(i) += 1.0 * i * values[j];
+    lsum += a(i);
+  }
+};
+
+struct Opts {
+  bool no_par_for         = false;
+  bool no_par_reduce      = false;
+  bool no_par_reduce_view = false;
+};
+
+template <int V>
+void run(int N, int M, int K, const Opts& opts) {
+  std::string l_no_fence, l_fence, l_red_no_fence, l_red_fence,
+      l_red_view_no_fence, l_red_view_fence;
+  {
+    std::ostringstream ostream;
+    ostream << "RunNoFence_" << N << "_" << K << std::endl;
+    l_no_fence = ostream.str();
+  }
+  {
+    std::ostringstream ostream;
+    ostream << "RunFence_" << N << "_" << K << std::endl;
+    l_fence = ostream.str();
+  }
+  {
+    std::ostringstream ostream;
+    ostream << "RunReduceNoFence_" << N << "_" << K << std::endl;
+    l_red_no_fence = ostream.str();
+  }
+  {
+    std::ostringstream ostream;
+    ostream << "RunReduceFence_" << N << "_" << K << std::endl;
+    l_red_fence = ostream.str();
+  }
+  {
+    std::ostringstream ostream;
+    ostream << "RunReduceViewNoFence_" << N << "_" << K << std::endl;
+    l_red_view_no_fence = ostream.str();
+  }
+  {
+    std::ostringstream ostream;
+    ostream << "RunReduceViewFence_" << N << "_" << K << std::endl;
+    l_red_view_fence = ostream.str();
+  }
+
+  double result;
+  Kokkos::View<double*> a("A", N);
+  Kokkos::View<double> v_result("result");
+  TestFunctor<V> f(a, K);
+  TestRFunctor<V> rf(a, K);
+  Kokkos::Timer timer;
+
+  // initialize to a obviously wrong value
+  double time_no_fence        = -1;  // launch loop
+  double time_no_fence_fenced = -1;  // launch loop then fence
+  double time_fence           = -1;  // launch&fence loop
+  double time_fence_fenced    = -1;  // launch&fence loop then fence
+
+  double time_red_no_fence        = -1;
+  double time_red_no_fence_fenced = -1;
+  double time_red_fence           = -1;
+  double time_red_fence_fenced    = -1;
+
+  double time_red_view_no_fence        = -1;
+  double time_red_view_no_fence_fenced = -1;
+  double time_red_view_fence           = -1;
+  double time_red_view_fence_fenced    = -1;
+
+  if (!opts.no_par_for) {
+    // warmup
+    Kokkos::parallel_for(l_no_fence, N, f);
+    Kokkos::parallel_for(l_no_fence, N, f);
+    Kokkos::parallel_for(l_no_fence, N, f);
+    Kokkos::parallel_for(l_no_fence, N, f);
```

Comment:
Any reason to unroll the loop?
```suggestion
    for (int i = 0; i < 4; ++i)
      Kokkos::parallel_for(l_no_fence, N, f);
```